c++/src/kj/compat/http.c++

```diff
@@ -2073,49 +2073,49 @@ public:
   }
 
   kj::Promise<void> writeBodyData(const void* buffer, size_t size) {
-    KJ_REQUIRE(!writeInProgress, "concurrent write()s not allowed") { return kj::READY_NOW; }
-    KJ_REQUIRE(inBody) { return kj::READY_NOW; }
+    KJ_REQUIRE(!writeInProgress, "concurrent write()s not allowed");
+    KJ_REQUIRE(inBody);
 
     writeInProgress = true;
     auto fork = writeQueue.fork();
     writeQueue = fork.addBranch();
 
-    return fork.addBranch().then([this,buffer,size]() {
-      return inner.write(buffer, size);
-    }).then([this]() {
-      writeInProgress = false;
-    });
+    co_await fork.addBranch();
+    co_await inner.write(buffer, size);
+
+    // TODO(cleanup): Use KJ_DEFER instead?
```

Comment:
No, these intentionally don't use KJ_DEFER. Perhaps add a comment:

> We intentionally don't use KJ_DEFER to clean this up because if an exception is thrown, we want to block further writes.

Reply:
Removed.